### PR TITLE
[clang] Additional FP classification functions

### PR DESCRIPTION
**Pull request by spavloff**
C language standard defined library functions `iszero`, `issignaling` and `issubnormal`, which did not have counterparts among clang builtin functions. This change adds new functions:

    __builtin_iszero
    __builtin_issubnormal
    __builtin_issignaling

They provide builtin implementation for the missing standard functions.

### CODE REVIEW

File: clang/include/clang/Basic/Builtins.def [hunk 1]
```diff
@@ -494,6 +494,9 @@ BUILTIN(__builtin_isinf,      "i.", "FnctE")
 BUILTIN(__builtin_isinf_sign, "i.", "FnctE")
 BUILTIN(__builtin_isnan,      "i.", "FnctE")
 BUILTIN(__builtin_isnormal,   "i.", "FnctE")
+BUILTIN(__builtin_issubnormal,"i.", "FnctE")
+BUILTIN(__builtin_iszero,     "i.", "FnctE")
+BUILTIN(__builtin_issignaling,"i.", "FnctE")
```
Comment (tbaederr) [inline clang/include/clang/Basic/Builtins.def:499]: The `E` means they can be evaluated at compile time, but I don't see changes to `ExprConstant.cpp`?

Comment (spavloff) [reply clang/include/clang/Basic/Builtins.def:499]: Yes, this is a copy-past error. Thanks for the catch.

**Comment by spavloff** [comment]
Ping.

**Review by kpneal — APPROVED** [review]
The rest LGTM.

**Comment by spavloff** [comment]
Thanks!

**Pull request merged**